File: 1582-SpecialPositionsInABinaryMatrix/SpecialPositionsInABinaryMatrix.java
import java.util.*;

class SpecialPositionsInABinaryMatrix {
    
    class Solution {
        public int numSpecial(int[][] mat) {
            List<Integer> specialCol = new ArrayList<>();
            
            for(int i = 0; i < mat.length; i++){
                int check = 0;
                int col = -1;
                for(int j = 0; j < mat[i].length; j++){
                    if(mat[i][j] == 1) {
                        check++;
                        col = j;
                    }
                }
                
                if (check == 1){
                    specialCol.add(col);
                }
            }
            
            int ans = 0;
            for(Integer j: specialCol){
                int check = 0;
                for(int i = 0; i < mat.length; i++){
                    if(mat[i][j] == 1) check++;
                }
                if (check == 1){
                    ans++;
                }
            }
                
            return ans;
        }
    }
}

File: 1582-SpecialPositionsInABinaryMatrix/SpecialPositionsInABinaryMatrix.go
package specialpositionsinabinarymatrix

func numSpecial(mat [][]int) int {
	specialCol := make([]int, 0)

	for i, _ := range mat {
		count := 0
		col := -1
		for j, _ := range mat[i] {
			if mat[i][j] == 1 {
				col = j
				count++
			}
		}

		if count == 1 {
			specialCol = append(specialCol, col)
		}
	}

	ans := 0
	for _, j := range specialCol {
		count := 0
		for i, _ := range mat {
			if mat[i][j] == 1 {
				count++
			}
		}

		if count == 1 {
			ans++
		}
	}

	return ans
}
